.github/workflows/ci.yml

```diff
@@ -0,0 +1,56 @@
+# This workflow uses actions that are not certified by GitHub.  They are
+# provided by a third-party and are governed by separate terms of service,
+# privacy policy, and support documentation.
+#
+# This workflow will install a prebuilt Ruby version, install dependencies, and
+# run tests and linters.
+name: "Ruby on Rails CI"
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    services:
```

Comment:
Please create Dockerfile and run commands from entrypoint.sh.
Then you can setup docker-compose.ci.yml file with postgres service.

It would be as close to prod conditions as possible.

Example:
https://github.com/po8klasie/po8klasie-api/blob/master/docker-compose.ci.yml
https://github.com/po8klasie/po8klasie-api/blob/master/.github/workflows/test-building.yml

Pls also notice the use of extracting image sha and placing it inside ${DOCKER_IMAGE} env var.

Reply:
I don't think that it matters very much right now. We aren't building a proper staging env and this level of mimicking an architcture that is boud to change in the future is a waste of time. Where the rails code is running or where the database is is not important to their result. Issues resulting from database connectivity will result in test failure. Building  a proper staging env once we deciede on production architecture will be worthwhile in the future but not now.

Reply:
@micorix can you see this comment?